docs/examples/python-custom-runtime/Python-Based-Custom-Runtime-with-Model-Stored-on-PVC.md

```diff
@@ -0,0 +1,278 @@
+
+# Deploy a model stored on a Persistent Volume Claim
+
+Let's use namespace `modelmesh-serving` here:
+
+```shell
+kubectl config set-context --current --namespace=modelmesh-serving
+```
+
+## 1. Create PV and PVC for storing model file
+
+```shell
+kubectl apply -f - <<EOF
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: my-models-pv
+spec:
+  capacity:
+    storage: 1Gi
+  accessModes:
+    - ReadWriteMany
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: ""
+  hostPath:
+    path: "/mnt/models"
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: "my-models-pvc"
+spec:
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 1Gi
+EOF
+```
+
+## 2. Create a pod to access the PVC
+
+```shell
+kubectl apply -f - <<EOF
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "pvc-access"
+spec:
+  containers:
+    - name: main
+      image: ubuntu
+      command: ["/bin/sh", "-ec", "sleep 10000"]
+      volumeMounts:
+        - name: "my-pvc"
+          mountPath: "/mnt/models"
+  volumes:
+    - name: "my-pvc"
+      persistentVolumeClaim:
+        claimName: "my-models-pvc"
+EOF
+```
+
+## 3. Store the model on this persistent volume
+
+The sample model file we used in this doc is `sklearn/mnist-svm.joblib`.
+
+```shell
+curl -sOL https://github.com/kserve/modelmesh-minio-examples/raw/main/sklearn/mnist-svm.joblib
+```
+
+Copy this model file to the `pvc-access` pod:
+
+```shell
+kubectl cp mnist-svm.joblib pvc-access:/mnt/models/
+```
+
+Verify the model exists on the persistent volume：
+
+```shell
+kubectl exec -it pvc-access -- ls -alr /mnt/models/
+
+# total 348
+# -rw-rw-r-- 1 1000 1000 344817 Mar 19 08:37 mnist-svm.joblib
+# drwxr-xr-x 1 root root   4096 Mar 19 08:34 ..
+# drwxr-xr-x 2 root root   4096 Mar 19 08:37 .
+```
+
+## 4. Configure ModelMesh Serving to use the persistent volume claim
+
+Create the `model-serving-config` ConfigMap with the setting allowAnyPVC: true:
+
+```shell
+kubectl apply -f - <<EOF
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: model-serving-config
+data:
+  config.yaml: |
+    allowAnyPVC: true
+EOF
+```
+
+Verify the configuration setting:
+
+```shell
+kubectl get cm "model-serving-config" -o jsonpath="{.data['config\.yaml']}"
+```
+
+# Implement the Python-based Custom Runtime on MLServer
+
+All of the necessary resources are contained in [custom-model](./custom-model), including the model code, and the Dockerfile.
+
+## 1. Implement the API of MLModel
+
+Both `load` and `predict` must be implemented to support this custom `ServingRuntime`. The code file [custom_model.py](./custom-model/custom_model.py) provides a simplified implementation of `CustomMLModel` for model `mnist-svm.joblib`. You can read more about it [here](https://github.com/kserve/modelmesh-serving/blob/main/docs/runtimes/mlserver_custom.md).
+
+## 2. Build the custom `ServingRuntime` image
```

Comment:
nit for consistency (I think I may have introduced this, sorry) 

```suggestion
## 2. Build the custom ServingRuntime image
```

Reply:
no problem, I added this, removed.